test/test_sox_compatibility.py

```diff
@@ -1,510 +1,304 @@
+    skipIfNoSoxBackend,
+    skipIfNoExec,
+    TempDirMixin,
+    TorchaudioTestCase,
+    get_asset_path,
+    sox_utils,
+    load_wav,
+    save_wav,
+    get_whitenoise,
+)
+
+
+@skipIfNoSoxBackend
+@skipIfNoExec('sox')
+class TestFunctionalFiltering(TempDirMixin, TorchaudioTestCase):
+    def run_sox_effect(self, input_file, effect):
+        output_file = self.get_temp_path('expected.wav')
+        sox_utils.run_sox_effect(input_file, output_file, [str(e) for e in effect])
+        return load_wav(output_file)
+
+    def assert_sox_effect(self, result, input_path, effects, atol=1e-04, rtol=1e-5):
+        expected, _ = self.run_sox_effect(input_path, effects)
+        self.assertEqual(result, expected, atol=atol, rtol=rtol)
+
+    def get_whitenoise(self, sample_rate=8000):
+        noise = get_whitenoise(
+            sample_rate=sample_rate, duration=3, scale_factor=0.9,
         )
-        self.noise_waveform, _ = torchaudio.load(self.noise_filepath, normalization=True)
+        path = self.get_temp_path("whitenoise.wav")
+        save_wav(path, noise, sample_rate)
+        return noise, path
 
     def test_gain(self):
-        test_filepath = common_utils.get_asset_path('steam-train-whistle-daniel_simon.wav')
-        waveform, _ = common_utils.load_wav(test_filepath)
-
-        waveform_gain = F.gain(waveform, 3)
-        self.assertTrue(waveform_gain.abs().max().item(), 1.)
-
-        E = torchaudio.sox_effects.SoxEffectsChain()
-        E.set_input_file(test_filepath)
-        E.append_effect_to_chain("gain", [3])
-        sox_gain_waveform = E.sox_build_flow_effects()[0]
-
-        self.assertEqual(waveform_gain, sox_gain_waveform, atol=1e-04, rtol=1e-5)
+        path = get_asset_path('steam-train-whistle-daniel_simon.wav')
+        data, _ = load_wav(path)
+        result = F.gain(data, 3)
+        self.assert_sox_effect(result, path, ['gain', 3])
 
     def test_dither(self):
-        test_filepath = common_utils.get_asset_path('steam-train-whistle-daniel_simon.wav')
-        waveform, _ = common_utils.load_wav(test_filepath)
-
-        waveform_dithered = F.dither(waveform)
-        waveform_dithered_noiseshaped = F.dither(waveform, noise_shaping=True)
+        path = get_asset_path('steam-train-whistle-daniel_simon.wav')
+        data, _ = load_wav(path)
+        result = F.dither(data)
+        self.assert_sox_effect(result, path, ['dither'])
 
-        E = torchaudio.sox_effects.SoxEffectsChain()
-        E.set_input_file(test_filepath)
-        E.append_effect_to_chain("dither", [])
-        sox_dither_waveform = E.sox_build_flow_effects()[0]
-
-        self.assertEqual(waveform_dithered, sox_dither_waveform, atol=1e-04, rtol=1e-5)
-        E.clear_chain()
-
-        E.append_effect_to_chain("dither", ["-s"])
-        sox_dither_waveform_ns = E.sox_build_flow_effects()[0]
-
-        self.assertEqual(waveform_dithered_noiseshaped, sox_dither_waveform_ns, atol=1e-02, rtol=1e-5)
-
-    def test_vctk_transform_pipeline(self):
```

Comment:
This test has been removed because it does not test anything new.

`channels` and `rate` effects are no effect because the input file is already one channel with 16k Hz, then the remaining effect is `gain -h gain -rh dither -s`, but applying gain twice does not make sense (`sox` command returns error) and `dither` is tested separately.